default.nix

```diff
@@ -1,48 +1,42 @@
-{ customConfig ? {}
-, withHoogle ? true
-, ... }:
-#
-# The default.nix file. This will generate targets for all
-# buildables (see release.nix for nomenclature, excluding
-# the "build machine" last part, specific to release.nix), eg.:
-#
-# - nix build -f default.nix nix-tools.tests.io-sim # All `io-sim` tests
-# - nix build -f default.nix nix-tools.tests.ouroboros-consensus.test-consensus
-#
-# Generated targets include anything from stack.yaml (via nix-tools:stack-to-nix and the nix/regenerate.sh script)
-# or cabal.project (via nix-tools:plan-to-nix), including all
-# version overrides specified there.
-#
-# Nix-tools stack-to-nix will generate the `nix/.stack-pkgs.nix`
-# file which is imported from the `nix/pkgs.nix` where further
-# customizations outside of the ones in stack.yaml/cabal.project
-# can be specified as needed for nix/ci.
-#
-# Please run `nix/regenerate.sh` after modifying stack.yaml
-# or relevant part of cabal configuration files.
-# When switching to recent stackage or hackage package version,
-# you might also need to update the iohk-nix common lib. You
-# can do so by running the `nix/update-iohk-nix.sh` script.
-#
-# More information about iohk-nix and nix-tools is available at:
-# https://github.com/input-output-hk/iohk-nix/blob/master/docs/nix-toolification.org#for-a-stackage-project
-#
-
-# We will need to import the iohk-nix common lib, which includes
-# the nix-tools tooling.
+{ system ? builtins.currentSystem
+, crossSystem ? null
+# allows to cutomize haskellNix (ghc and profiling, see ./nix/haskell.nix)
+, config ? {}
+# allows to override dependencies of the project without modifications,
+# eg. to test build against local checkout of nixpkgs and iohk-nix:
+# nix build -f default.nix cardano-ledger --arg sourcesOverride '{
+#   iohk-nix = ../iohk-nix;
+# }'
+, sourcesOverride ? {}
+# pinned version of nixpkgs augmented with overlays (iohk-nix and our packages).
+, pkgs ? import ./nix { inherit system crossSystem config sourcesOverride; }
+, gitrev ? pkgs.iohkNix.commitIdFromGitRepoOrZero ./.git
+}:
+with pkgs; with commonLib;
```

Comment:
where can we find instructions how to build various parts with nix?
Did the build instruction changed (if that the case we'll need to update some READMEs).